src/ast/semantic_analyser.cpp

```diff
@@ -165,6 +173,25 @@ ProbeType SemanticAnalyser::single_provider_type(void)
   return type;
 }
 
+AddrSpace SemanticAnalyser::find_addrspace(ProbeType pt)
+{
+  switch (pt)
+  {
+    case ProbeType::kprobe:
+    case ProbeType::kretprobe:
+    case ProbeType::kfunc:
+    case ProbeType::kretfunc:
+    case ProbeType::tracepoint:
+      return AddrSpace::kernel;
+    case ProbeType::uprobe:
+    case ProbeType::uretprobe:
+    case ProbeType::usdt:
+      return AddrSpace::user;
+    default:
+      return AddrSpace::none;
```

Comment:
should we abort here in case someone adds a new type but doesn't update this?

Reply:
Agree.

Reply:
The compiler should be able to catch it for us (no `default` case to force exhaustive switch). I think we had some weird `-W` settings on that prevented doing that before. Should look again.

Reply:
b20320b028ff is in now so we should drop the `default` here

Reply:
ok. Will remove it.

Reply:
Kept AddrSpace::none for now. See tools/runqlen.bt, builtin.ctx runtime test, i:ms:1 (struct x*)ctx)->x.